src/renderer/screens/Balances/components/NetworkBalances/NetworkBalances.tsx

```diff
@@ -70,20 +50,29 @@ const NetworkBalances = ({
       return acc;
     }, {}) || {};
 
-  const filteredAssets = chain.assets.filter((asset) => {
-    if (query) {
-      return (
-        includes(asset.symbol, query) ||
-        (!searchSymbolOnly && (includes(chain.name, query) || includes(asset.name, query)))
-      );
-    }
-
-    const balance = balancesObject[asset.assetId];
-
-    return (
-      !hideZeroBalance || !balance || balance.verified !== true || (balance && totalAmount(balance) !== ZERO_BALANCE)
-    );
-  });
+  const filteredAssets = useMemo(
```

Comment:
`const balances = getLiveNetworkBalances(accountIds, chain.chainId);`
It gives us new array instance every time balances are being updated in DB which is fine
But here comes `useMemo` 😄
Memo will store all those different `balances` with all related filter results that actually won't be any helpful because we can't receive same `balances` instance again so the memo cannot provide already calculated value.